lib/internal/tls.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const net = require('net');
+
+function canonicalIp(address) {
+  // Convert the ip address into the same format
+  // stored in certificates
+  if (net.isIPv6(address)) {
+    const b = ['0', '0', '0', '0', '0', '0', '0', '0'];
+
+    const s = address.split('::');
+    if (s.length === 2) {
```

Comment:
Is there actually any IPv6 without the two dots?

Reply:
Yes. The :: notation is just an abbreviation for any number of zero-bytes. There can very well be an IPv6-address without any zero-bytes. And the address comes from the Host-header, so it might be that the http-client requested an address with the IPv6 address spelled out, e.g. https://[0.0.0.0.0.0.0.1] for localhost.